docs/reference/api.md

```diff
@@ -0,0 +1,77 @@
+# API
+
+Pebble exposes an API to allow remote clients to interact with the daemon. The API can start and stop services, add configuration layers to the plan, and so on.
+
+## Accessing the API
```

Comment:
On reflection, maybe we can just drop this section heading -- this whole doc is a reference for "accessing the API".

However, we do need a section something like "structure of the API" where we mention the different parts of the API in general terms: URL with path or query params, request, response, JSON format, request types (sync, async).